src/components/main-view/main-view.jsx

```diff
@@ -1,69 +1,69 @@
-      director: {
-        name: "Lana Wachowski, Lilly Wachowski",
-        bio: "The Wachowskis are American film and TV directors, writers, and producers.",
-        birthYear: { $numberInt: "1965" },
-        deathYear: null
-      },
-      rating: { $numberDouble: "8.7" },
-      imageURL: "https://example.com/matrix.jpg",
-      featured: true,
-      actors: [{ $oid: "66f1fd9db3b2033c0957ef1e" }, { $oid: "66f1fd9db3b2033c0957ef1f" }]
-    },
-    {
-      id: 3,
-      title: "Interstellar",
-      description: "A team of explorers travel through a wormhole in space in an attempt to ensure humanity's survival.",
-      genre: { name: "Sci-Fi", description: "Science fiction films often explore futuristic concepts." },
-      director: {
-        name: "Christopher Nolan",
-        bio: "Christopher Nolan is a British-American filmmaker known for making thought-provoking, visually stunning films.",
-        birthYear: { $numberInt: "1970" },
-        deathYear: null
-      },
-      rating: { $numberDouble: "8.6" },
-      imageURL: "https://example.com/interstellar.jpg",
-      featured: true,
-      actors: [{ $oid: "66f1fd9db3b2033c0957ef20" }, { $oid: "66f1fd9db3b2033c0957ef21" }]
-    }
-  ]);
-
+  const [movies, setMovies] = useState([]); 
+  
   const [selectedMovie, setSelectedMovie] = useState(null);
 
- 
+  useEffect(() => {
+    fetch("https://flixhive-cf7fbbd939d2.herokuapp.com/movies")
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
+        const moviesFromApi = data.map((doc =>{
+          return {
+            id: doc._id,
+            title: doc.title,
+            description: doc.description,
+            genre: {
+              name: doc.genre.name,
+              description: doc.genre.description,
+            },
+            director: {
+              name: doc.director.name,
+              bio: doc.director.bio,
+              birthYear: doc.director.birthYear,
+              deathYear: doc.director.deathYear,
+            },
+            imageURL: doc.imageURL,
+            rating: doc.rating,
+            featured: doc.featured,
+            actors: doc.actors
+          }
+        }))
+       setMovies(moviesFromApi);
+      })
+      .catch((error) => {
+        console.error("Error fetching movies:", error);
+      });
+    }, []);
+ //get similar movies based on genre, excluding the movie that is currently being viewed
+    const getSimilarMovies = (genre) => {
+      return movies.filter((movie) => movie.genre.name === genre.name &&movie.id !== selectedMovie.id);
+    };
+    
      if (selectedMovie) {
-      return <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />;
-     
-    }
+      const similarMovies = getSimilarMovies(selectedMovie.genre);
+      return (
+        <div>
+      <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />;
```

Comment:
Hi @cinchronicity , 
The extra semicolon was introduced in this line. So you can just get rid of it. It's only the purely javascript statements that end with it. The JSX(html-like) syntax is always wrapped around parentheses.()
```suggestion
      <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />
```